x11-packages/wlroots/0006-fix-vulkan-renderer-build.patch

```diff
@@ -0,0 +1,277 @@
+new file mode 100644
+index 00000000..f93c8845
+--- /dev/null
++++ b/render/vulkan/shaders/texture.frag.h
+@@ -0,0 +1,30 @@
++	// 1114.1.0
++	 #pragma once
++const uint32_t texture_frag_data[] = {
++	0x07230203,0x00010000,0x0008000b,0x0000001f,0x00000000,0x00020011,0x00000001,0x0006000b,
++	0x00000001,0x4c534c47,0x6474732e,0x3035342e,0x00000000,0x0003000e,0x00000000,0x00000001,
++	0x0007000f,0x00000004,0x00000004,0x6e69616d,0x00000000,0x00000009,0x00000011,0x00030010,
++	0x00000004,0x00000007,0x00030003,0x00000002,0x000001c2,0x00040005,0x00000004,0x6e69616d,
++	0x00000000,0x00050005,0x00000009,0x5f74756f,0x6f6c6f63,0x00000072,0x00030005,0x0000000d,
++	0x00786574,0x00030005,0x00000011,0x00007675,0x00030005,0x00000015,0x004f4255,0x00050006,
++	0x00000015,0x00000000,0x68706c61,0x00000061,0x00040005,0x00000017,0x61746164,0x00000000,
++	0x00040047,0x00000009,0x0000001e,0x00000000,0x00040047,0x0000000d,0x00000022,0x00000000,
++	0x00040047,0x0000000d,0x00000021,0x00000000,0x00040047,0x00000011,0x0000001e,0x00000000,
++	0x00050048,0x00000015,0x00000000,0x00000023,0x00000050,0x00030047,0x00000015,0x00000002,
++	0x00020013,0x00000002,0x00030021,0x00000003,0x00000002,0x00030016,0x00000006,0x00000020,
++	0x00040017,0x00000007,0x00000006,0x00000004,0x00040020,0x00000008,0x00000003,0x00000007,
++	0x0004003b,0x00000008,0x00000009,0x00000003,0x00090019,0x0000000a,0x00000006,0x00000001,
++	0x00000000,0x00000000,0x00000000,0x00000001,0x00000000,0x0003001b,0x0000000b,0x0000000a,
++	0x00040020,0x0000000c,0x00000000,0x0000000b,0x0004003b,0x0000000c,0x0000000d,0x00000000,
++	0x00040017,0x0000000f,0x00000006,0x00000002,0x00040020,0x00000010,0x00000001,0x0000000f,
++	0x0004003b,0x00000010,0x00000011,0x00000001,0x0004002b,0x00000006,0x00000013,0x00000000,
++	0x0003001e,0x00000015,0x00000006,0x00040020,0x00000016,0x00000009,0x00000015,0x0004003b,
++	0x00000016,0x00000017,0x00000009,0x00040015,0x00000018,0x00000020,0x00000001,0x0004002b,
++	0x00000018,0x00000019,0x00000000,0x00040020,0x0000001a,0x00000009,0x00000006,0x00050036,
++	0x00000002,0x00000004,0x00000000,0x00000003,0x000200f8,0x00000005,0x0004003d,0x0000000b,
++	0x0000000e,0x0000000d,0x0004003d,0x0000000f,0x00000012,0x00000011,0x00070058,0x00000007,
++	0x00000014,0x0000000e,0x00000012,0x00000002,0x00000013,0x0003003e,0x00000009,0x00000014,
++	0x00050041,0x0000001a,0x0000001b,0x00000017,0x00000019,0x0004003d,0x00000006,0x0000001c,
++	0x0000001b,0x0004003d,0x00000007,0x0000001d,0x00000009,0x0005008e,0x00000007,0x0000001e,
++	0x0000001d,0x0000001c,0x0003003e,0x00000009,0x0000001e,0x000100fd,0x00010038
++};
+diff --git a/render/vulkan/vulkan.c b/render/vulkan/vulkan.c
+index 61f49ea5..307eec81 100644
+--- a/render/vulkan/vulkan.c
++++ b/render/vulkan/vulkan.c
+@@ -26,6 +26,7 @@ static bool check_extension(const VkExtensionProperties *avail,
+ 	return false;
+ }
+ 
++#if !defined (__ANDROID__) && !defined (__TERMUX__)
+ static VkBool32 debug_callback(VkDebugUtilsMessageSeverityFlagBitsEXT severity,
```

Comment:
Seems useful, why was this disabled too?

Reply:
It fails to build on arm32

Reply:
The following patch may fix the build issue in arm32.

<details><summary>Click here to see the diff</summary>

```diff
--- a/render/vulkan/vulkan.c
+++ b/render/vulkan/vulkan.c
@@ -22,7 +22,7 @@
 	return false;
 }
 
-static VkBool32 debug_callback(VkDebugUtilsMessageSeverityFlagBitsEXT severity,
+static VKAPI_ATTR VkBool32 debug_callback(VkDebugUtilsMessageSeverityFlagBitsEXT severity,
 		VkDebugUtilsMessageTypeFlagsEXT type,
 		const VkDebugUtilsMessengerCallbackDataEXT *debug_data,
 		void *data) {
```

</details>

I can try to upstream the change if necessary.

Reply:
> The following patch may fix the build issue in arm32.
> 
> Click here to see the diff
> I can try to upstream the change if necessary.

It works, thank you.

Reply:
Ah calling convention, nice catch.

Reply:
Not calling convention, it's attribute.

```c
#elif defined(__ANDROID__) && defined(__ARM_ARCH) && __ARM_ARCH >= 7 && defined(__ARM_32BIT_STATE)
#define VKAPI_ATTR __attribute__((pcs("aapcs-vfp")))
```

See `$PREFIX/include/vulkan/vk_platform.h`

Reply:
I have created a merge request to fix it in upstream https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4641

Reply:
It's already merged, that was extremely fast.